### PR TITLE
feat: add directory method

### PR DESCRIPTION
**Pull request by sveisvei**
## JIRA Issue
[COREWEB-83](https://jira.finn.no/browse/COREWEB-83)

## Status
**READY**

## Description
Add a new async method for listing files in a directory.

It returns an array with objects of shape: `{fileName, content}`

## Todos
- [x] Tests
- [x] Documentation

## Related PRs
* https://github.com/asset-pipe/asset-pipe-sink-fs/pull/40
* https://github.com/asset-pipe/asset-pipe-sink-mem/pull/48

### CODE REVIEW

**Comment by SimenB** [comment]
?

**Comment by sveisvei** [comment]
@SimenB dont know how it closed... anyway - added some code to normalize dirnames etc - feedback/suggestions welcome.

File: lib/sink.js [hunk 1]
```diff
@@ -159,6 +160,63 @@ module.exports = class SinkGCS extends EventEmitter {
         return !!(result && result[0]);
     }
 
+    _getPrefixes(directoryName) {
```
Comment (trygve-lie) [inline lib/sink.js]: Symbols for privacy?

File: lib/sink.js [hunk 2]
```diff
@@ -159,6 +160,63 @@ module.exports = class SinkGCS extends EventEmitter {
...
+        if (directoryName === '/' || !directoryName) {
+            filterPrefix = '.';
+        } else {
+            prefix = `${directoryName.replace(/^\//, '').replace(/\/$/, '')}/`;
+            filterPrefix = prefix;
+        }
+        return { filterPrefix, prefix };
+    }
+
+    _getPrefixFilterIterator(filterPrefix) {
```
Comment (trygve-lie) [inline lib/sink.js]: Symbols for privacy?

Comment (sveisvei) [reply lib/sink.js]: Sure!

File: lib/sink.js (hunk 1 above)
Comment (sveisvei) [reply lib/sink.js]: Yes!

File: lib/sink.js (hunk 2 above)
Comment (SimenB) [reply lib/sink.js]: None of them touch `this`, so I'd just lift them out of the `class` itself

Comment (sveisvei) [reply lib/sink.js]: Sure @SimenB. Will add symbol to `_getGcsFile`.

Comment (sveisvei) [reply lib/sink.js]: done, moved 2 fucntions to scope - and made one method private.

**Comment by sveisvei** [comment]
I did not close this... happened when I pushed :(

**Comment by trygve-lie** [comment]
Nop. I don't have access to that...

**Comment by trygve-lie** [comment]
Think you need to push a change on the branch to reopen it.

**Comment by SimenB** [comment]
the branch has zero commits, did you do a `git reset` and force push without commiting?

**Comment by sveisvei** [comment]
<- (akward)

**Pull request merged**